DOCSP-43074: Specify a Query (#30)
mongodb/docs-c: source/read/specify-a-query.txt

.. _c-specify-query:

===============
Specify a Query
===============

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 2
   :class: singlecol

.. facet::
   :name: genre
   :values: reference

.. meta::
   :keywords: expressions, operations, read, write, filter

Overview
--------

In this guide, you can learn how to specify a query by using the {+driver-short+}.

> You can refine the set of documents that a query returns by creating a
> **query filter**. A query filter is an expression that specifies the search

Reviewer: ```suggestion
You can define a **query filter** to retrieve specific documents from a collection
when performing a query. A query filter is an expression that specifies the search
```

criteria MongoDB uses to match documents in a read or write operation.

> In a query filter, you can prompt the driver to search for documents with an
> exact match to your query, or you can compose query filters to express more
> complex matching criteria.

Reviewer: ```suggestion
By defining a query filter, you can direct the driver to search for documents with an
exact match to your query, or you can compose query filters to express more
complex matching criteria.
```

Sample Data
~~~~~~~~~~~

The examples in this guide run operations on a collection called
``fruits`` that contains the following documents:

.. code-block:: json

   { "_id": 1, "name": "apples", "quantity": 5, "rating": 3, "color": "red", "type": ["fuji", "honeycrisp"] },
   { "_id": 2, "name": "bananas", "quantity": 7, "rating": 4, "color": "yellow", "type": ["cavendish"] },
   { "_id": 3, "name": "oranges", "quantity": 6, "rating": 2, "type": ["naval", "mandarin"] },
   { "_id": 4, "name": "pineapple", "quantity": 3, "rating": 5, "color": "yellow" },

The following code example shows how to create a database and collection, then
insert the sample documents into your collection:

.. literalinclude:: /includes/read/specify-a-query.c
   :start-after: start-sample-data
   :end-before: end-sample-data
   :language: c
   :dedent:

>    :copyable:

Reviewer: Q: Should we follow convention used in other pages/standardizations to use the Atlas sample data when possible instead of using generated data?

Author: Can change to the restaurants database.
